### ISSUE TITLE
Qt error when Remove selected then Remove all rows in EngDiff UI table

### ISSUE THREAD

**Issue by RichardWaiteSTFC** [mantidproject/mantid#33244]
**Describe the bug**
Found during manual testing for 6.3 release #33191 
Qt error when Remove Selected then Remove all rows in EngDiff UI table
```
Error occurred in handler:
Traceback (most recent call last):
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\fitting_ads_observer.py", line 20, in wrapper
func(*args, **kwargs)
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\fitting_ads_observer.py", line 75, in replaceHandle
self.replace_callback(name, workspace)
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\data_handling\data_widget.py", line 40, in replace_workspace
self.presenter.replace_workspace(name, workspace)
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\data_handling\data_presenter.py", line 105, in replace_workspace
self._repopulate_table()
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\data_handling\data_presenter.py", line 139, in _repopulate_table
self._remove_all_table_rows()
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\data_handling\data_presenter.py", line 290, in _remove_all_table_rows
self.view.remove_all()
File "C:\MantidNightlyInstall\bin\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\data_handling\data_view.py", line 189, in remove_all
self.table_selection.setRowCount(0)
RuntimeError: wrapped C/C++ object of type QTableWidget has been deleted
```
**To Reproduce**
Having trouble reproducing...
(1) Load a few (2-3?) focussed spectra into the fitting tab of EngDiff UI
(2) Click a row in the table and click button Remove Selected
(3) Click Remove All button

**Expected behavior**
No exception, table in UI would be cleared and the `_logs` group of workspaces in ADS will be deleted.

**Platform/Version (please complete the following information):**
 - Nightly 6.2 (4 Jan) windows

**Comment by RichardWaiteSTFC** [comment]
Could be related #33220

**Comment by DannyHindson** [comment]
I can't reproduce this on Windows. I think I have seen this error though occasionally while doing development work on this UI but haven't been able to pin down the actions that cause it

**Comment by RichardWaiteSTFC** [comment]
Perhaps the most reproducible way of triggering the same/similar error is in #33281?

**Comment by martyngigg** [comment]
Fixed by #33299
